src/components/DynamicFormContainer/DynamicFormLayout.tsx

```diff
@@ -1,53 +1,131 @@
+    nextForms.splice(nextForms.length - 1, 1);
+    setForms(nextForms);
     setActiveFormIndex(undefined);
   };
 
+  const onBackToFormSelection = (): void => {
+    removeCurrentForm();
+  };
+
+  const onNewForm = (): void => {
+    if (!validateCurrentForm()) setActiveFormIndex(undefined);
+  };
+
+  const onFormSubmit = (): void => {
+    if (!validateCurrentForm()) setIsSubmitted(true);
+  };
+
+  const closeDeleteModal = (): void => {
+    setDeleteModal(false);
+  };
+
+  const deleteForm = (): void => {
+    removeCurrentForm();
+    closeDeleteModal();
+  };
+
   return (
     <Container>
-      <div className="container mx-auto">
-        <div
-          onClick={onBackToFormSelection}
-          className="text-grey flex cursor-pointer py-4 w-20"
-          data-testid="back-button"
-        >
-          <SvgIcon>
-            <SvgIconArrowLeft />
-          </SvgIcon>
-          <div className="pl-2">Back</div>
-        </div>
-        <ProgressBar step={2} />
-        <Title>Fill and Preview Form</Title>
-      </div>
+      <DeleteModal
+        deleteForm={deleteForm}
+        show={showDeleteModal}
+        closeDeleteModal={closeDeleteModal}
+      />
+      <DynamicFormHeader
+        onBackToFormSelection={onBackToFormSelection}
+        onNewForm={onNewForm}
+        onFormSubmit={onFormSubmit}
+      />
       <div className="bg-white-dark p-6">
         <div className="bg-white container mx-auto p-4">
-          <div className="text-grey-dark flex items-center">
-            <div className="align-middle">Preview mode:</div>
-            <ToggleSwitch
-              isOn={isPreviewMode}
-              handleToggle={() => setIsPreviewMode(!isPreviewMode)}
-            />
+          <div className="flex justify-between">
+            <div className="text-grey-dark flex items-center">
+              <div className="align-middle">Preview mode:</div>
+              <ToggleSwitch
+                isOn={isPreviewMode}
+                handleToggle={() => setIsPreviewMode(!isPreviewMode)}
+              />
+            </div>
+            <Button data-testid="delete-button" onClick={() => setDeleteModal(true)}>
+              <div className="rounded w-12 h-12 border border-solid border-lightgrey flex items-center justify-center">
+                <SvgIcon className="text-lightgrey-dark">
+                  <SvgIconTrash />
+                </SvgIcon>
+              </div>
+            </Button>
           </div>
-          <div className="">
-            <DynamicForm form={activeForm} handleSubmit={console.log} />
+          {formError && formError.length > 0 && (
+            <div className="bg-red-lighter rounded max-w-screen-sm mx-auto h-12 flex items-center">
+              <SvgIcon className="text-red mx-3">
+                <SvgIconXCircle />
+              </SvgIcon>
+              <div className="text-red text-xl text-center">
+                This form has errors. Please fix the errors and submit again.
```

Comment:
Don't you want to spell the errors out?

Reply:
yup, will display a list of errors upon validation